metric/instrument/asyncfloat64.go

```diff
@@ -39,6 +39,46 @@ type Float64Observable interface {
 // Warning: methods may be added to this interface in minor releases.
 type Float64ObservableCounter interface{ Float64Observable }
 
+// Float64ObservableCounterConfig contains options for asynchronous counter
+// instruments that record int64 values.
+type Float64ObservableCounterConfig struct {
+	description string
+	unit        string
+	callbacks   []Float64Callback
+}
+
+// NewFloat64ObservableCounterConfig returns a new
+// [Float64ObservableCounterConfig] with all opts applied.
+func NewFloat64ObservableCounterConfig(opts ...Float64ObservableCounterOption) Float64ObservableCounterConfig {
```

Comment:
Not directly connected with this PR, but I prefer asking here before creating an issue.

Why do we export the config types and their factory function (e.g. `NewFloat64ObserverConfig` and `Float64ObservableCounterConfig`)?

_I am asking here before creating an issue, because I might be missing some initial design decision. However, I do not understand the benefit of exporting it and  it feels like it is against the recommendation from https://github.com/open-telemetry/opentelemetry-go/blob/main/CONTRIBUTING.md#config_

Reply:
Yeah, good question for sure. We export here because it is the API and we know the implementations (i.e. the SDK) will need to directly create and use the configs. If we didn't export them the options with their private method could not configure something accessible to an external user.

Reply:
Thank you.